### PR TITLE
#316 Implemented namespaces in application.

### PR DESCRIPTION
**Pull request by ceesvanegmond**
See #356. I think namespacing the helper functions will remove alot of problems. Especially those who's using the Laravel/Lumen framework.

### CODE REVIEW

File: recipe/codeigniter.php [hunk 1]
```diff
@@ -5,18 +5,20 @@
  * file that was distributed with this source code.
  */
 
+use Deployer\Functions;
```
Comment (elfet) [inline recipe/codeigniter.php]: Why not this?
```php
namespace Deployer\Functions;
```

File: src/functions.php [hunk 2]
```diff
@@ -1,4 +1,7 @@
 <?php
+
+namespace Deployer\Functions;
```
Comment (elfet) [inline src/functions.php]: Must be under file comment.

**Comment by elfet** [comment]
Will add this to v4.

**Comment by Baldinof** [comment]
Hi, why not putting `functions.php` under `Deployer` namespace, instead of `Deployer\Functions`.

So we'll just have to put `namespace Deployer;`  in `deploy.php` and recipes files, which is sexier than `namespace Deployer\Functions;`

It could be another cool namespace :smile:

**Comment by JHGitty** [comment]
I think everything should be under ``Deployer`` namespace directly.

**Comment by elfet** [comment]
Agree.

**Comment by JHGitty** [comment]
ping @ceesvanegmond
Can you fix it or should we create a new PR?

**Comment by elfet** [comment]
PR?

**Comment by JHGitty** [comment]
PR = Pull Request.

**Comment by Baldinof** [comment]
I think this PR could be closed as 5eadb57 implements namespace for functions.

**Comment by elfet** [comment]
Right!

**Pull request closed without merging**